census/templates/census/event_form.html

```diff
@@ -11,11 +11,17 @@
         <section class='tablet:grid-col-8 tablet:grid-offset-2 section-card padding-x-4 padding-y-2 tablet:margin-4'>
 
             <form method="post" class="">
+
+            {% if form.is_private_event.value and not user.is_authenticated %}
```

Comment:
I think a re-direct may be a better solution if someone tries to access a private event and is not authenticated.

This can be a follow-up improvement, it would keep the whole template nested 1 level less (although there is still more that can be done to make this template more readable)